### ISSUE TITLE
feat[dev]: ✨ implement A11y Re-Improvements to sd-flag in teaser template (with chip and flag)

### ISSUE THREAD

**Issue by yoezlem** [solid-design-system/solid#2077]
## Description
This tasks describes the Figma implementation for: #2048 

This task describes the Storybook implementation for the sd-flag component as part of the A11y re-improvements outlined in Epic #2048.

A11y improvements for sd-flag can be found here: [sd-flag Notes](https://www.notion.so/matuzo/sd-flag-1baf3f6e1bc08102b9a1cbefb9f6170f).

------------------------------------------

## Comment
/

~## New Notes~
~- The contrast ratio in `sd-flag--neutral-500` is high enough (8:1) according to WCAG but practically it’s no well readable.~

## Notes
- The heading in a teaser should be the first item in the DOM. A heading introduces a new thematic region and separates the following content from the previous region. **Although chips and flags are visually the first items in the teaser, they should come after the heading in the DOM.**

## DoR
- [ ] Item has been estimated by the team
- [x] Item dependencies have been identified and documented

## DoD
- [ ] Documentation has been created/updated (if applicable)
- [ ] Migration Guide has been created/updated (if applicable)
- [ ] Relevant E2E tests (Features, A11y, Bug fixes) are created/updated
- [ ] Relevant stories (Features, A11y) are created/updated
- [ ] Implementation works successfully on `feature` branch
